// asana/dist/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ApiClient", {
  enumerable: true,
  get: function get() {
    return _ApiClient["default"];
  }
});
Object.defineProperty(exports, "AttachmentApi", {
  enumerable: true,
  get: function get() {
    return _AttachmentApi["default"];
  }
});
Object.defineProperty(exports, "AttachmentResponse", {
  enumerable: true,
  get: function get() {
    return _AttachmentResponse["default"];
  }
});
Object.defineProperty(exports, "AttachmentResponseData", {
  enumerable: true,
  get: function get() {
    return _AttachmentResponseData["default"];
  }
});
Object.defineProperty(exports, "AttachmentResponseDataParent", {
  enumerable: true,
  get: function get() {
    return _AttachmentResponseDataParent["default"];
  }
});
Object.defineProperty(exports, "ErrorResponse", {
  enumerable: true,
  get: function get() {
    return _ErrorResponse["default"];
  }
});
Object.defineProperty(exports, "ErrorResponseErrorsInner", {
  enumerable: true,
  get: function get() {
    return _ErrorResponseErrorsInner["default"];
  }
});
var _ApiClient = _interopRequireDefault(require("./ApiClient"));
var _AttachmentResponse = _interopRequireDefault(require("./model/AttachmentResponse"));
var _AttachmentResponseData = _interopRequireDefault(require("./model/AttachmentResponseData"));
var _AttachmentResponseDataParent = _interopRequireDefault(require("./model/AttachmentResponseDataParent"));
var _ErrorResponse = _interopRequireDefault(require("./model/ErrorResponse"));
var _ErrorResponseErrorsInner = _interopRequireDefault(require("./model/ErrorResponseErrorsInner"));
var _AttachmentApi = _interopRequireDefault(require("./api/AttachmentApi"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
