// asana/dist/model/AttachmentResponseDataParent.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
/**
 * The AttachmentResponseDataParent model module.
 * @module model/AttachmentResponseDataParent
 * @version 0.0.1
 */
var AttachmentResponseDataParent = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>AttachmentResponseDataParent</code>.
   * The task this attachment is attached to.
   * @alias module:model/AttachmentResponseDataParent
   */
  function AttachmentResponseDataParent() {
    _classCallCheck(this, AttachmentResponseDataParent);
    AttachmentResponseDataParent.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(AttachmentResponseDataParent, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>AttachmentResponseDataParent</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/AttachmentResponseDataParent} obj Optional instance to populate.
     * @return {module:model/AttachmentResponseDataParent} The populated <code>AttachmentResponseDataParent</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new AttachmentResponseDataParent();
        if (data.hasOwnProperty('gid')) {
          obj['gid'] = _ApiClient["default"].convertToType(data['gid'], 'String');
        }
        if (data.hasOwnProperty('resource_type')) {
          obj['resource_type'] = _ApiClient["default"].convertToType(data['resource_type'], 'String');
        }
        if (data.hasOwnProperty('name')) {
          obj['name'] = _ApiClient["default"].convertToType(data['name'], 'String');
        }
        if (data.hasOwnProperty('resource_subtype')) {
          obj['resource_subtype'] = _ApiClient["default"].convertToType(data['resource_subtype'], 'String');
        }
      }
      return obj;
    }
  }]);
  return AttachmentResponseDataParent;
}();
/**
 * The task this attachment is attached to.
 * @member {String} gid
 */
AttachmentResponseDataParent.prototype['gid'] = undefined;

/**
 * The base type of this resource.
 * @member {String} resource_type
 */
AttachmentResponseDataParent.prototype['resource_type'] = undefined;

/**
 * The name of the task.
 * @member {String} name
 */
AttachmentResponseDataParent.prototype['name'] = undefined;

/**
 * The resource subtype of the parent resource that the filter applies to.
 * @member {String} resource_subtype
 */
AttachmentResponseDataParent.prototype['resource_subtype'] = undefined;
var _default = AttachmentResponseDataParent;
exports["default"] = _default;
