// asana/dist/model/ErrorResponseErrorsInner.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;
var _ApiClient = _interopRequireDefault(require("../ApiClient"));
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }
function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }
function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, _toPropertyKey(descriptor.key), descriptor); } }
function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); Object.defineProperty(Constructor, "prototype", { writable: false }); return Constructor; }
function _toPropertyKey(arg) { var key = _toPrimitive(arg, "string"); return _typeof(key) === "symbol" ? key : String(key); }
function _toPrimitive(input, hint) { if (_typeof(input) !== "object" || input === null) return input; var prim = input[Symbol.toPrimitive]; if (prim !== undefined) { var res = prim.call(input, hint || "default"); if (_typeof(res) !== "object") return res; throw new TypeError("@@toPrimitive must return a primitive value."); } return (hint === "string" ? String : Number)(input); }
/**
 * The ErrorResponseErrorsInner model module.
 * @module model/ErrorResponseErrorsInner
 * @version 0.0.1
 */
var ErrorResponseErrorsInner = /*#__PURE__*/function () {
  /**
   * Constructs a new <code>ErrorResponseErrorsInner</code>.
   * @alias module:model/ErrorResponseErrorsInner
   */
  function ErrorResponseErrorsInner() {
    _classCallCheck(this, ErrorResponseErrorsInner);
    ErrorResponseErrorsInner.initialize(this);
  }

  /**
   * Initializes the fields of this object.
   * This method is used by the constructors of any subclasses, in order to implement multiple inheritance (mix-ins).
   * Only for internal use.
   */
  _createClass(ErrorResponseErrorsInner, null, [{
    key: "initialize",
    value: function initialize(obj) {}

    /**
     * Constructs a <code>ErrorResponseErrorsInner</code> from a plain JavaScript object, optionally creating a new instance.
     * Copies all relevant properties from <code>data</code> to <code>obj</code> if supplied or a new instance if not.
     * @param {Object} data The plain JavaScript object bearing properties of interest.
     * @param {module:model/ErrorResponseErrorsInner} obj Optional instance to populate.
     * @return {module:model/ErrorResponseErrorsInner} The populated <code>ErrorResponseErrorsInner</code> instance.
     */
  }, {
    key: "constructFromObject",
    value: function constructFromObject(data, obj) {
      if (data) {
        obj = obj || new ErrorResponseErrorsInner();
        if (data.hasOwnProperty('message')) {
          obj['message'] = _ApiClient["default"].convertToType(data['message'], 'String');
        }
        if (data.hasOwnProperty('help')) {
          obj['help'] = _ApiClient["default"].convertToType(data['help'], 'String');
        }
        if (data.hasOwnProperty('phrase')) {
          obj['phrase'] = _ApiClient["default"].convertToType(data['phrase'], 'String');
        }
      }
      return obj;
    }
  }]);
  return ErrorResponseErrorsInner;
}();
/**
 * Message providing more detail about the error that occurred, if available.
 * @member {String} message
 */
ErrorResponseErrorsInner.prototype['message'] = undefined;

/**
 * Additional information directing developers to resources on how to address and fix the problem, if available.
 * @member {String} help
 */
ErrorResponseErrorsInner.prototype['help'] = undefined;

/**
 * *500 errors only*. A unique error phrase which can be used when contacting developer support to help identify the exact occurrence of the problem in Asana’s logs.
 * @member {String} phrase
 */
ErrorResponseErrorsInner.prototype['phrase'] = undefined;
var _default = ErrorResponseErrorsInner;
exports["default"] = _default;
